controllers/front/OrderController.php

```diff
@@ -146,24 +146,9 @@ protected function bootstrap()
     protected function saveDataToPersist(CheckoutProcess $process)
     {
         $data = $process->getDataToPersist();
-        $addressValidator = new AddressValidator($this->context);
-        $customer = $this->context->customer;
         $cart = $this->context->cart;
 
-        $shouldGenerateChecksum = false;
-
-        if ($customer->isGuest()) {
-            $shouldGenerateChecksum = true;
-        } else {
-            $invalidAddressIds = $addressValidator->validateCartAddresses($cart);
-            if (empty($invalidAddressIds)) {
-                $shouldGenerateChecksum = true;
-            }
-        }
-
-        $data['checksum'] = $shouldGenerateChecksum
-            ? $this->cartChecksum->generateChecksum($cart)
-            : null;
+        $data['checksum'] = $this->cartChecksum->generateChecksum($cart);
```

Comment:
Hi @lartist 

just a question regarding your modification, previously the Address was being checked to be sure it is valid before updating the cart checksum. With your modification it's not done anymore so although it means we're sure the checksum is updated more often, doesn't it also mean that it can be updated with an invalid address?